$NetBSD: patch-CMakeLists.txt,v 1.2 2026/10/27 20:18:32 wiz Exp $

Don't attempt to deal with link-time optimization manually; CMake is smart
enough to figure it out on its own.

Serialize building of the packs, c4groups re-uses the same names.

--- CMakeLists.txt.orig	2018-03-16 17:25:33.000000000 +0000
+++ CMakeLists.txt
@@ -91,38 +91,6 @@ endfunction()
 	endforeach()
 endfunction()
 
-CHECK_CXX_COMPILER_FLAG("-flto" USE_GCC_STYLE_LTCG)
-if(USE_GCC_STYLE_LTCG)
-	set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -flto")
-	set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL} -flto")
-	add_linker_flags(optimized MODULES exe shared FLAGS -flto)
-
-	# Use GCC's ar and ranlib wrappers if necessary, because the plain ones 
-	# don't understand lto objects without an explicit plugin parameter
-	if(CMAKE_C_COMPILER MATCHES "gcc$")
-		set(LTCG_NEEDS_AR_WRAPPER 1)
-		set(LTCG_AR_WRAPPER_PREFIX "${CMAKE_C_COMPILER}")
-	elseif(CMAKE_C_COMPILER MATCHES "cc$")
-		set(LTCG_NEEDS_AR_WRAPPER 1)
-		set(LTCG_AR_WRAPPER_PREFIX "gcc")
-	else()
-		set(LTCG_NEEDS_AR_WRAPPER 0)
-	endif()
-
-	if(LTCG_NEEDS_AR_WRAPPER)
-		find_program(AR_WRAPPER "${LTCG_AR_WRAPPER_PREFIX}-ar")
-		if (AR_WRAPPER)
-			message("Using ${AR_WRAPPER} instead of ${CMAKE_AR} to support lto objects.")
-			set(CMAKE_AR "${AR_WRAPPER}" CACHE FILEPATH "Path to an ar that supports lto objects." FORCE)
-		endif()
-		find_program(RANLIB_WRAPPER "${LTCG_AR_WRAPPER_PREFIX}-ranlib")
-		if (RANLIB_WRAPPER)
-			message("Using ${RANLIB_WRAPPER} instead of ${CMAKE_RANLIB} to support lto objects.")
-			set(CMAKE_RANLIB "${RANLIB_WRAPPER}" CACHE FILEPATH "Path to a ranlib that supports lto objects." FORCE)
-		endif()
-	endif()
-endif()
-
 if(MSVC)
 	if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
 		# Set up some additional flags only if we're using the MS compiler,
@@ -1510,6 +1478,7 @@ endif()
 	list(APPEND OC_C4GROUPS Music.ocg)
 endif()
 
+set(OC_C4GROUPS_PREV "")
 foreach(group ${OC_C4GROUPS})
 	if (APPLE)
 		if (CMAKE_GENERATOR STREQUAL Xcode)
@@ -1518,7 +1487,7 @@ foreach(group ${OC_C4GROUPS})
 				"$<TARGET_FILE:${native_c4group}>"
 				"${CMAKE_CURRENT_SOURCE_DIR}/planet/${group}"
 				# leave out third parameter here so the script can figure out Xcode-ish paths as usual
-				DEPENDS "${native_c4group}"
+				DEPENDS "${native_c4group}" ${OC_C4GROUPS_PREV}
 			)
 		else()
 			add_custom_command(TARGET openclonk
@@ -1526,7 +1495,7 @@ foreach(group ${OC_C4GROUPS})
 				"$<TARGET_FILE:${native_c4group}>"
 				"${CMAKE_CURRENT_SOURCE_DIR}/planet/${group}"
 				"${CMAKE_CURRENT_BINARY_DIR}/openclonk.app/Contents/Resources"
-				DEPENDS "${native_c4group}"
+				DEPENDS "${native_c4group}" ${OC_C4GROUPS_PREV}
 			)
 		endif()
 	else()
@@ -1534,11 +1503,12 @@ foreach(group ${OC_C4GROUPS})
 			OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${group}
 			COMMAND "${native_c4group}" ARGS "${CMAKE_CURRENT_SOURCE_DIR}/planet/${group}" -t "${CMAKE_CURRENT_BINARY_DIR}/${group}"
 			MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/planet/${group}
-			DEPENDS "${native_c4group}"
+			DEPENDS "${native_c4group}" ${OC_C4GROUPS_PREV}
 			VERBATIM
 		)
 		install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${group} DESTINATION share/games/openclonk)
 	endif()
+	set(OC_C4GROUPS_PREV "${CMAKE_CURRENT_BINARY_DIR}/${group}")
 endforeach()
 
 if(NOT APPLE)
